nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/JsonNodeStatusHistoryDumpFactory.java

```diff
@@ -27,6 +29,7 @@
 
     @Override
     public StatusHistoryDump create(int days) {
+        Preconditions.checkArgument(days > 0, String.format("The number of days shall be greater than 0. The current value is %s.", days));
```

Comment:
The `%s` is for string input primarilty. Format documentation suggests '%d' for integer values